Bubble sort array

cmchan04 opened the pull request:
Bubble sort for array, the code is still running on my side currently. This PR is just for tracking for this moment

cmchan04 committed:
Bubble sort for array and code run testing (not yet tested)

cmchan04 committed:
Show sample records

cmchan04 committed:
Merge branch 'main' of https://github.com/cmchan04/DSTR_Assignment into bubble_sort_array

BengRhui commented:
Looks good as for now. @cmchan04 once the sort finishes do inform me here. I'll add some comments b4 approving haha.
[I forgot to include location in my methods oops, maybe we shall deal with it later once we finishes all our code.]

cmchan04 committed:
added a counter and commented the code that runs the bubble sort

cmchan04 commented:
> Looks good as for now. @cmchan04 once the sort finishes do inform me here. I'll add some comments b4 approving haha. [I forgot to include location in my methods oops, maybe we shall deal with it later once we finishes all our code.]

@BengRhui  I terminated the program ald. Took me 40 mins without knowing it is running probably or not QQ

cmchan04 committed:
Release memory

cmchan04 commented:
> Looks good as for now. @cmchan04 once the sort finishes do inform me here. I'll add some comments b4 approving haha. [I forgot to include location in my methods oops, maybe we shall deal with it later once we finishes all our code.]

 @BengRhui  You can continue modify lah I go for other sorting first

cmchan04 committed:
Insertion sort

cmchan04 commented:
I have added the insertion sort in this branch too, forgot to branch out jn.

BengRhui committed:
Change swap() to static

I've discovered that this method is located inside the Transaction class, but not the array class (mixed it up with my Linked List class XD), so we had no choice but to declare it as a static method since we're not creating a class for arrays.

BengRhui committed:
Clean up bubble sort algorithm

The algorithm is slightly modified so that it can compare type -> if type matches then we compare ID. Also added a boolean to end the bubble sort early if no operations are performed throughout the outer iteration.

BengRhui committed:
Update insertion sort

Some comments are added for insertion sort, and the comparison logic is updated to arrange the transactions better.

BengRhui reviewed (approved):
- I've slightly modified some parts of the insertion sort algorithm (the final boolean check is not accurate).
- Added some comments for the codes. 
- Most importantly, the comparison logic is updated to make sure comparisons are clear for the algorithm to run.

BengRhui commented:
All in all, everything looks good, just minor changes only. @cmchan04 if everything looks good on your side you can merge this branch into main ya
